flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/util/AwsV2Util.java

```diff
@@ -0,0 +1,237 @@
+		final SdkAsyncHttpClient httpClient = createHttpClient(config, NettyNioAsyncHttpClient.builder());
+		final ClientOverrideConfiguration overrideConfiguration = createClientOverrideConfiguration(config, ClientOverrideConfiguration.builder());
+		final KinesisAsyncClientBuilder clientBuilder = KinesisAsyncClient.builder();
+
+		return createKinesisAsyncClient(configProps, clientBuilder, httpClient, overrideConfiguration);
+	}
+
+	@VisibleForTesting
+	static SdkAsyncHttpClient createHttpClient(
+			final ClientConfiguration config,
+			final NettyNioAsyncHttpClient.Builder httpClientBuilder) {
+		httpClientBuilder
+			.maxConcurrency(config.getMaxConnections())
+			.connectionTimeout(Duration.ofMillis(config.getConnectionTimeout()))
+			.writeTimeout(Duration.ofMillis(config.getSocketTimeout()))
+			.connectionMaxIdleTime(Duration.ofMillis(config.getConnectionMaxIdleMillis()))
+			.useIdleConnectionReaper(config.useReaper());
+
+		if (config.getConnectionTTL() > -1) {
+			httpClientBuilder.connectionTimeToLive(Duration.ofMillis(config.getConnectionTTL()));
+		}
+
+		return httpClientBuilder.build();
+	}
+
+	@VisibleForTesting
+	static ClientOverrideConfiguration createClientOverrideConfiguration(
+			final ClientConfiguration config,
+			final ClientOverrideConfiguration.Builder overrideConfigurationBuilder) {
+
+		overrideConfigurationBuilder
+			.putAdvancedOption(SdkAdvancedClientOption.USER_AGENT_PREFIX, AWSUtil.formatFlinkUserAgentPrefix())
+			.putAdvancedOption(SdkAdvancedClientOption.USER_AGENT_SUFFIX, config.getUserAgentSuffix());
+
+		if (config.getRequestTimeout() > 0) {
+			overrideConfigurationBuilder.apiCallAttemptTimeout(Duration.ofMillis(config.getRequestTimeout()));
+		}
+
+		if (config.getClientExecutionTimeout() > 0) {
+			overrideConfigurationBuilder.apiCallTimeout(Duration.ofMillis(config.getClientExecutionTimeout()));
+		}
+
+		return overrideConfigurationBuilder.build();
+	}
+
+	@VisibleForTesting
+	static KinesisAsyncClient createKinesisAsyncClient(
+			final Properties configProps,
+			final KinesisAsyncClientBuilder clientBuilder,
+			final SdkAsyncHttpClient httpClient,
+			final ClientOverrideConfiguration overrideConfiguration) {
+
+		if (configProps.containsKey(AWSConfigConstants.AWS_ENDPOINT)) {
+			final URI endpointOverride = URI.create(configProps.getProperty(AWSConfigConstants.AWS_ENDPOINT));
+			clientBuilder.endpointOverride(endpointOverride);
+		}
+
+		return clientBuilder
+			.httpClient(httpClient)
+			.overrideConfiguration(overrideConfiguration)
+			.credentialsProvider(getCredentialsProvider(configProps))
+			.region(getRegion(configProps))
+			.build();
+	}
+
+	/**
+	 * Return a {@link AWSCredentialsProvider} instance corresponding to the configuration properties.
+	 *
+	 * @param configProps the configuration properties
+	 * @return The corresponding AWS Credentials Provider instance
+	 */
+	public static AwsCredentialsProvider getCredentialsProvider(final Properties configProps) {
```

Comment:
Why should we overload this method? Will there be an alternative prefix?

Reply:
Yes when using `AssumeRole` it is called from line 205 with a different prefix.